src/Akismet.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Akismet;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\GuzzleException;
+use Psr\Http\Message\ResponseInterface;
+
+class Akismet
+{
+    public $isConfigured = true;
+    private $apiUrl;
+    private $flarumVersion;
+    private $extensionVersion;
+
+    private $params = [];
+    public $proTip;
+
+    public function __construct(string $apiKey, string $homeUrl, $flarumVersion, $extensionVersion, $inDebugMode = false)
+    {
+        if (empty($apiKey)) {
+            return $this->isConfigured = false;
+        }
+
+        $this->apiUrl = "https://$apiKey.rest.akismet.com/1.1";
+        $this->setBlog($homeUrl);
+
+        $this->flarumVersion = $flarumVersion;
+        $this->extensionVersion = $extensionVersion;
+
+        if ($inDebugMode) {
+            $this->setTest();
+        }
+    }
+
+    /**
+     * @param  string  $type  eg. submit-spam, submit-spam or submit-ham;
+     * @throws GuzzleException
+     */
+    public function sendRequest(string $type): ResponseInterface
+    {
+        $client = new Client();
+        return $client->request('POST', "$this->apiUrl/$type", [
+            'headers'     => [
+                'User-Agent' => "Flarum/$this->flarumVersion | Akismet/$this->extensionVersion",
+            ],
+            'form_params' => $this->params,
+        ]);
+    }
+
+    /**
+     * @throws GuzzleException
+     */
+    public function isSpam(): bool
+    {
+        $response = $this->sendRequest('comment-check');
+
+        if ($response->hasHeader('X-akismet-pro-tip')) {
+            $this->proTip = $response->getHeaderLine('X-akismet-pro-tip');
+        }
+
+        if ($response->getBody()->getContents() === 'true') {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @throws GuzzleException
+     */
+    public function submitSpam()
+    {
+        $this->sendRequest('submit-spam');
+    }
+
+    /**
+     * @throws GuzzleException
+     */
+    public function submitHam()
+    {
+        $this->sendRequest('submit-ham');
+    }
+
+    /**
+     * The front page or home URL of the instance making the request. For a blog or wiki this would be the front page. Note: Must be a full URI, including http://.
+     */
+    public function setBlog(string $url): Akismet
```

Comment:
A common pattern in fluent APIs like this is keeping the object immutable past construction, and instead returning cloned copies. So `setX` would be `withX`. This might be a bit trickier here since the params are in an array, not directly on the object though.

Reply:
Something like this?
```php
    public function withBlog(string $url): Akismet
    {
        $clone = clone $this;
        $clone->params['blog'] = $url;
        return $clone;
    }
```

Reply:
Done in https://github.com/flarum/akismet/pull/24/commits/6003cfea2e0842f46f82b1292b0e908eb4b0081e